fix NoSuchElementException in ViolationsUtil on fresh run

```
Caused by: java.util.NoSuchElementException
        at com.netflix.nebula.archrules.gradle.ViolationsUtil$Companion.printSummary(ViolationsUtil.kt:59)
        at com.netflix.nebula.archrules.gradle.ViolationsUtil.printSummary(ViolationsUtil.kt)
```

--- a/nebula-archrules-gradle-plugin/src/main/kotlin/com/netflix/nebula/archrules/gradle/ArchrulesRunnerPlugin.kt
+++ b/nebula-archrules-gradle-plugin/src/main/kotlin/com/netflix/nebula/archrules/gradle/ArchrulesRunnerPlugin.kt
@@ -36,13 +36,17 @@ class ArchrulesRunnerPlugin : Plugin<Project> {
                 }
             val checkTasks = project.tasks.withType<CheckRulesTask>()
             val jsonReportTask = project.tasks.register<PrintJsonReportTask>("archRulesJsonReport") {
-                getDataFiles().set(project.fileTree(archRulesReportDir).filter { it.extension == "data" })
+                getDataFiles().set(
+                    project.provider { (project.tasks.withType<CheckRulesTask>().flatMap { it.outputs.files }) }
+                )
                 getJsonReportFile().set(archRulesReportDir.map { it.file("report.json").asFile })
                 dependsOn(checkTasks)
             }
 
             val consoleReportTask = project.tasks.register<PrintConsoleReportTask>("archRulesConsoleReport") {
-                getDataFiles().set(project.fileTree(archRulesReportDir).filter { it.extension == "data" })
+                getDataFiles().set(
+                    project.provider { (project.tasks.withType<CheckRulesTask>().flatMap { it.outputs.files }) }
+                )
                 dependsOn(checkTasks)
                 onlyIf { archRulesExt.consoleReportEnabled.get() }
             }

--- a/nebula-archrules-gradle-plugin/src/test/kotlin/com/netflix/nebula/archrules/gradle/ArchrulesRunnerPluginTest.kt
+++ b/nebula-archrules-gradle-plugin/src/test/kotlin/com/netflix/nebula/archrules/gradle/ArchrulesRunnerPluginTest.kt
@@ -2,6 +2,7 @@ package com.netflix.nebula.archrules.gradle
 
 import nebula.test.dsl.*
 import nebula.test.dsl.TestKitAssertions.assertThat
+import org.gradle.kotlin.dsl.named
 import org.gradle.testfixtures.ProjectBuilder
 import org.gradle.testkit.runner.TaskOutcome
 import org.junit.jupiter.api.Test
@@ -24,6 +25,21 @@ class ArchrulesRunnerPluginTest {
         project.plugins.apply(ArchrulesRunnerPlugin::class.java)
         val configuration = project.configurations.findByName("archRules")
         assertThat(configuration).isNotNull
+    }
+
+    @Test
+    fun `report inputs are correct`() {
+        val project = ProjectBuilder.builder().build()
+        project.plugins.apply("java")
+        project.plugins.apply(ArchrulesRunnerPlugin::class.java)
+        val consoleReport = project.tasks.named<PrintConsoleReportTask>("archRulesConsoleReport")
+        assertThat(consoleReport.get().dataFiles.get())
+            .`as`("console report inputs are correct")
+            .hasSize(2)
+        val jsonReport = project.tasks.named<PrintJsonReportTask>("archRulesJsonReport")
+        assertThat(jsonReport.get().dataFiles.get())
+            .`as`("json report inputs are correct")
+            .hasSize(2)
     }
 
     @ParameterizedTest

